homeassistant/components/bedrock_agent/__init__.py

```diff
@@ -0,0 +1,156 @@
+
+        modelId = self.entry.data[CONST_MODEL_ID]
+        body = json.dumps({"prompt": question})
+
+        # switch case statement
+        if modelId.startswith("amazon.titan-text-express-v1"):
+            body = json.dumps(
+                {
+                    "inputText": question,
+                    "textGenerationConfig": {
+                        "temperature": 0,
+                        "topP": 1,
+                        "maxTokenCount": 512,
+                    },
+                }
+            )
+        elif modelId.startswith("anthropic.claude"):
+            body = json.dumps(
+                {
+                    "prompt": f"\n\nHuman:{question}\n\nAssistant:",
+                    "max_tokens_to_sample": 200,
+                    "temperature": 0.1,
+                    "top_p": 0.9,
+                }
+            )
+        elif modelId.startswith("ai21.j2"):
+            body = json.dumps(
+                {
+                    "prompt": question,
+                    "temperature": 0.5,
+                    "topP": 0.5,
+                    "maxTokens": 200,
+                    "countPenalty": {"scale": 0},
+                    "presencePenalty": {"scale": 0},
+                    "frequencyPenalty": {"scale": 0},
+                }
+            )
+        elif modelId.startswith("mistral.mistral-"):
+            body = json.dumps(
+                {
+                    "prompt": f"<s>[INST] {question} [/INST]",
+                    "max_tokens": 512,
+                    "temperature": 0.5,
+                    "top_p": 0.9,
+                    "top_k": 50,
+                }
+            )
+        # elif modelId == "meta.llama2-":
+        #     return "meta.llama2-"
+
+        accept = "application/json"
+        contentType = "application/json"
+
+        bedrock_response = await self.hass.async_add_executor_job(
+            partial(
+                self.bedrock.invoke_model,
+                body=body,
+                modelId=modelId,
+                accept=accept,
+                contentType=contentType,
+            ),
+        )
+
+        response_body = json.loads(bedrock_response.get("body").read())
+        if modelId.startswith("amazon.titan-text-express-v1"):
+            answer = response_body["results"][0]["outputText"]
+        elif modelId.startswith("anthropic.claude"):
+            answer = response_body["completion"]
+        elif modelId.startswith("ai21.j2"):
+            answer = response_body["completions"][0]["data"]["text"]
+        elif modelId.startswith("mistral.mistral-"):
+            answer = response_body["outputs"][0]["text"]
+        else:
+            answer = "Sorry I am not able to understand my underlying model."
+
+        return answer
+
+    async def async_process(
+        self, user_input: agent.ConversationInput
+    ) -> agent.ConversationResult:
+        """Process a sentence."""
+        answer = await self.async_call_bedrock(user_input.text)
```

Comment:
Would it be possible to support chat history with `user_input.conversation_id` as a key?

Reply:
AFAIK there is currently no model that supports this feature. Models that support a conversation like Claude do this via prompt chaining. Prompt chaining support will definitely be added in a later version.